src/metabase/lib/drill_thru.cljc

```diff
@@ -96,12 +109,13 @@
            (when (lib.metadata/editable? query)
              (let [{:keys [query stage-number]} (lib.query/wrap-native-query-with-mbql
                                                  query stage-number (:card-id context))
-                   dim-contexts                 (dimension-contexts context)]
+                   context'                     (context-with-dimensions-or-row-dimensions query context)
+                   dim-contexts                 (dimension-contexts context')]
                (for [{:keys [f return-drills-for-dimensions?]} available-drill-thru-fns
-                     context                                   (if (and return-drills-for-dimensions? dim-contexts)
+                     context''                                 (if (and return-drills-for-dimensions? dim-contexts)
                                                                  dim-contexts
-                                                                 [context])
-                     :let                                      [drill (f query stage-number context)]
+                                                                 [context'])
+                     :let                                      [drill (f query stage-number context'')]
```

Comment:
We don't generally do this `foo'` style unless it's really complex, and then it's better to give the different versions more descriptive names anyway. I think just using `context` here is fine.